addons/binding/org.openhab.binding.russound/src/main/java/org/openhab/binding/russound/internal/discovery/RioSystemDiscovery.java

```diff
@@ -0,0 +1,188 @@
+            while (inetAddresses.hasMoreElements()) {
+                final InetAddress inetAddress = inetAddresses.nextElement();
+
+                if (inetAddress instanceof Inet6Address) {
+                    continue;
+                }
+
+                final String hostAddress = inetAddress.getHostAddress();
+                final int lastDot = hostAddress.lastIndexOf('.');
+                final String subnet = hostAddress.substring(0, lastDot + 1);
+                logger.trace("Starting scan of {}", subnet);
+
+                final String[] addresses = new String[254];
+                for (int i = 1; i < 255; i++) {
+                    addresses[i - 1] = subnet + i;
+                }
+                for (int i = 0; i < addresses.length; i++) {
+                    final int j = i;
+                    scheduler.execute(new Runnable() {
+                        @Override
+                        public void run() {
+                            scanAddress(addresses[j]);
+                        }
+                    });
+                }
+            }
+        }
+    }
+
+    /**
+     * Helper method to scan a specific address. Will open up port 9621 on the address and if opened, query for any
+     * controller type (all 6 controllers are tested). If a valid type is found, a discovery result will be created.
+     *
+     * @param ipAddress a possibly null, possibly empty ip address (null/empty addresses will be ignored)
+     */
+    private void scanAddress(String ipAddress) {
+
+        if (StringUtils.isEmpty(ipAddress)) {
+            return;
+        }
+
+        final SocketSession session = new SocketChannelSession(ipAddress, RioConstants.RioPort);
+        try {
+            final WaitingSessionListener listener = new WaitingSessionListener();
+            session.addListener(listener);
+            session.connect(100);
+            logger.debug("Connected to port {}:{} - testing to see if RIO", ipAddress, RioConstants.RioPort);
+
+            // can't check for system properties because DMS responds to those -
+            // need to check if any controllers are defined
+            for (int c = 1; c < 7; c++) {
+                session.sendCommand("GET C[" + c + "].type");
+                final String resp = listener.getResponse();
+                if (resp == null) {
+                    continue;
+                }
+                if (!resp.startsWith("S C[" + c + "].type=\"")) {
+                    continue;
+                }
+                final String type = resp.substring(13, resp.length() - 1);
+                if (!StringUtils.isBlank(type)) {
+                    logger.debug("Found a RIO type #{}", type);
+                    addResult(ipAddress, type);
+                    break;
+                }
+            }
+        } catch (Exception e) {
+            logger.trace("Connection couldn't be established to port {}:{}", ipAddress, RioConstants.RioPort);
```

Comment:
Change to catch only IOExceptions and trace is appropriate here since we are going to get a lot of messages for IP addresses that aren't valid (ie IOExceptions since the port on the address isn't available).